mailpoet/lib/Settings/SettingsChangeHandler.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace MailPoet\Settings;
+
+use MailPoet\Cron\Workers\InactiveSubscribers;
+use MailPoet\Cron\Workers\WooCommerceSync;
+use MailPoet\Entities\ScheduledTaskEntity;
+use MailPoet\Newsletter\Sending\ScheduledTasksRepository;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoetVendor\Carbon\Carbon;
+
+class SettingsChangeHandler {
```

Comment:
I didn't move everything over to this class, and @lysyjan I'm wondering which other methods, if any, are worth bringing over. Some things seemed a little less straightforward, like how `updateReEngagementEmailStatus` comes last and potentially alters the response of the API request.

I don't want to refactor this and leave it in a half-and-half state, but I did want to double check to see if you think some of the change logic needs to stay in `\MailPoet\API\JSON\v1\Settings`.

Reply:
@johnolek I'm okay with the current state. It was a good step for using Migrator and the new class as DI dependencies. I wouldn't want to complicate more this pull request.